### DISCUSSION TITLE
CIDFlatten for non cid-keyed fonts

### DISCUSSION THREAD

**Discussion by dnesscarkey** [fontforge/fontforge#4728] [Q&A] [answered]
Hi,

I am using a script that converts fonts to ttf and woff. Also, i want to add support for CID keyed fonts. However, when doing so, it is unable to convert non CID keyed fonts. Is there any way where we can use CIDFlatten for CID fonts and avoid it when the font source is not CID ?

```
Open($1,1)
CIDFlatten()
Generate($2+'.ttf')
Generate($2+'.woff')
Close()
```

**Comment by skef** [comment]
I'm not sure there's a specific way to do this in the native scripting language but doesn't FontsInFile() return multiple answers for a CID font? If so then you can use that and SizeOf to branch.

**Reply by dnesscarkey** [reply]
Thank you so much for your reply.

FontsInFile just returns the Font name and SizeOf returning just 1 for any type.

**Comment by jtanx** [comment, marked as answer]
https://fontforge.org/docs/scripting/python/fontforge.html#fontforge.font.cidsubfontnames

Use the python api

**Reply by dnesscarkey** [reply]
Thanks you @jtanx :)

**Reply by ctrlcctrlv** [reply]
Yeah, that's also the right move in general. The native scripting language is deprecated. We add a lot more to Python than to it.
